File: web/src/layout/RootXml.ts
//
// RootXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId } from 'khrysalis/dist/views/html'
import { customViewSetDelegate } from 'khrysalis/dist/views/CustomView.actual'
import htmlForDefault from './root.html'
//! Declares com.lightningkite.khrysaliswebsite.layouts.RootXml
export class RootXml {
    xmlRoot!: HTMLElement;
    rootStackHolder!: HTMLDivElement;
    dialogStackHolder!: HTMLDivElement;
    loadHtmlString(): string {
        return htmlForDefault;
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString());
        this.xmlRoot = view
        this.rootStackHolder = getViewById<HTMLDivElement>(view, "rootStackHolder");
        this.dialogStackHolder = getViewById<HTMLDivElement>(view, "dialogStackHolder");
        return view
    }
}

File: android/src/main/java/com/lightningkite/khrysaliswebsite/layouts/RootXml.kt
//
// RootXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.khrysaliswebsite.layouts

import android.widget.*
import android.view.*
import com.lightningkite.khrysalis.views.*
import com.lightningkite.khrysaliswebsite.R

class RootXml {

    lateinit var rootStackHolder: com.lightningkite.khrysalis.views.android.SwapView
    lateinit var dialogStackHolder: com.lightningkite.khrysalis.views.android.SwapView
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ViewDependency): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.root, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        rootStackHolder = view.findViewById<com.lightningkite.khrysalis.views.android.SwapView>(R.id.rootStackHolder)
        dialogStackHolder = view.findViewById<com.lightningkite.khrysalis.views.android.SwapView>(R.id.dialogStackHolder)
        
        
        
        return view
    }
}